bindings/java/src/org/sleuthkit/datamodel/HostManager.java

```diff
@@ -208,99 +213,106 @@ public Host createHost(String name) throws TskCoreException  {
+			}
 		} finally {
 			db.releaseSingleUserCaseWriteLock();
 		}
 	}
-	
+
 	/**
-	 * Get host with given name. 
-	 * 
-	 * @param name Host name to look for. 
+	 * Get host with given name.
+	 *
+	 * @param name        Host name to look for.
 	 * @param transaction Database transaction to use.
-	 * 
-	 * @return Optional with host.  Optional.empty if no matching host is found.
-	 * 
-	 * @throws TskCoreException 
+	 *
+	 * @return Optional with host. Optional.empty if no matching host is found.
+	 *
+	 * @throws TskCoreException
 	 */
-	public Optional<Host> getHost(String name, CaseDbTransaction transaction ) throws TskCoreException {
-		
+	public Optional<Host> getHost(String name, CaseDbTransaction transaction) throws TskCoreException {
+
 		return getHost(name, transaction.getConnection());
 	}
-	
+
 	/**
-	 * Get host with given name. 
-	 * 
-	 * @param name Host name to look for. 
+	 * Get host with given name.
+	 *
+	 * @param name       Host name to look for.
 	 * @param connection Database connection to use.
-	 * 
-	 * @return Optional with host.  Optional.empty if no matching host is found.
-	 * 
-	 * @throws TskCoreException 
+	 *
+	 * @return Optional with host. Optional.empty if no matching host is found.
+	 *
+	 * @throws TskCoreException
 	 */
-	private Optional<Host> getHost(String name, CaseDbConnection connection ) throws TskCoreException {
-		
-		String queryString = "SELECT * FROM tsk_hosts"
-							+ " WHERE LOWER(name) = LOWER('" + name + "')";
-		try (
-				Statement s = connection.createStatement();
-				ResultSet rs = connection.executeQuery(s, queryString)) {
+	private Optional<Host> getHost(String name, CaseDbConnection connection) throws TskCoreException {
 
-			if (!rs.next()) {
-				return Optional.empty();	// no match found
-			} else {
-				return Optional.of(new Host(rs.getLong("id"), rs.getString("name"), Host.HostStatus.fromID(rs.getInt("status"))));
+		String queryString = "SELECT * FROM tsk_hosts"
+				+ " WHERE LOWER(name) = LOWER(?)";
+		try (PreparedStatement s = connection.getPreparedStatement(queryString, Statement.RETURN_GENERATED_KEYS)) {
+			s.clearParameters();
+			s.setString(1, name);
+			
+			try (ResultSet rs = s.executeQuery()) {
+				if (!rs.next()) {
+					return Optional.empty();	// no match found
+				} else {
+					return Optional.of(new Host(rs.getLong("id"), rs.getString("name"), Host.HostStatus.fromID(rs.getInt("status"))));
+				}
 			}
 		} catch (SQLException ex) {
 			throw new TskCoreException(String.format("Error getting host with name = %s", name), ex);
 		}
 	}
-	
+
 	/**
 	 * Get all hosts.
-	 * 
+	 *
 	 * @return Collection of hosts.
-	 * @throws TskCoreException 
+	 *
+	 * @throws TskCoreException
 	 */
-	public Set<Host> getHosts() throws TskCoreException {
-		String queryString = "SELECT * FROM tsk_hosts";
+	public List<Host> getHosts() throws TskCoreException {
+		String queryString = "SELECT * FROM tsk_hosts WHERE status = ? ";
```

Comment:
If we're only going to return active hosts, is there any reason to have the status as a parameter? It doesn't seem like it needs to be a prepared statement either way. Also the top comment block should be changed to indicate that only active hosts will be returned.